src/libraries/FullMath.sol

```diff
@@ -109,9 +109,8 @@ library FullMath {
     function mulDivRoundingUp(uint256 a, uint256 b, uint256 denominator) internal pure returns (uint256 result) {
         unchecked {
             result = mulDiv(a, b, denominator);
-            if (mulmod(a, b, denominator) > 0) {
-                require(result < type(uint256).max);
-                result++;
+            if (mulmod(a, b, denominator) != 0) {
+                require(++result > 0);
```

Comment:
looks good. I see that the test `test_fuzz_mulDivRoundingUp` isn't very good because it checks 
```
assertTrue(result == numerator / d || result == numerator / d + 1);
```
🥲 
please could you update it to correctly identify whether it should have `+1` or not with a mod?

Reply:
Updated.